### ISSUE TITLE
[opencv4] build failure

### ISSUE THREAD

**Issue by nickanthony-dgl** [microsoft/vcpkg#38557]
### Operating system

Windows 10

### Compiler

MSVC

### Steps to reproduce the behavior

```Shell
When installing opencv4 without default features (`vcpkg install opencv4[core]`) it will fail during configuration because CMake will try to find flatbuffers. Since flatbuffers is not a dependency of opencv4 when default features are not enabled it will not be found and configuration will fail.

This issue occurs on version 4.8.0, port versions: 16-18. Port version 15 appears to work correctly.

Probably because the "fix-no-flatbuffers" patch was introduced in version 16
```

### Failure logs

This shows the configuration failure when building on the most recent port version (18):
[config-x64-win-llvm-release-out (2).log](https://github.com/microsoft/vcpkg/files/15205637/config-x64-win-llvm-release-out.2.log)

**Comment by FrankXie05** [comment]
@nickanthony-dgl Thanks for posting this issue, teh default functionality ensures that basic functionality is provided when the library is installed, making it easier for users to start using the library without worrying about missing necessary functionality. 
https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/concepts/default-features.md

**Comment by nickanthony-dgl** [comment]
So you're telling me it's intentional opencv4 should fail to build without default features?

**Comment by dg0yt** [comment]
> So you're telling me it's intentional opencv4 should fail to build without default features?

It must not fail without default features.

**Comment by nickanthony-dgl** [comment]
Fixed by #38594
